homeassistant/components/withings/__init__.py

```diff
@@ -39,68 +26,11 @@
     get_data_manager_by_webhook_id,
     json_message_response,
 )
-from .const import CONF_USE_WEBHOOK, CONFIG, LOGGER
+from .const import CONF_USE_WEBHOOK, LOGGER
 
 DOMAIN = const.DOMAIN
 PLATFORMS = [Platform.BINARY_SENSOR, Platform.SENSOR]
 
-CONFIG_SCHEMA = vol.Schema(
-    {
-        DOMAIN: vol.All(
-            cv.deprecated(const.CONF_PROFILES),
-            cv.deprecated(CONF_CLIENT_ID),
-            cv.deprecated(CONF_CLIENT_SECRET),
-            vol.Schema(
-                {
-                    vol.Optional(CONF_CLIENT_ID): vol.All(cv.string, vol.Length(min=1)),
-                    vol.Optional(CONF_CLIENT_SECRET): vol.All(
-                        cv.string, vol.Length(min=1)
-                    ),
-                    vol.Optional(const.CONF_USE_WEBHOOK, default=False): cv.boolean,
-                    vol.Optional(const.CONF_PROFILES): vol.All(
-                        cv.ensure_list,
-                        vol.Unique(),
-                        vol.Length(min=1),
-                        [vol.All(cv.string, vol.Length(min=1))],
-                    ),
-                }
-            ),
-        )
-    },
-    extra=vol.ALLOW_EXTRA,
-)
-
-
-async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
```

Comment:
We should have a proper deprecation (with raising an issue).
Then from there you can update your config entry with the YAML setting for webhook so it updates it and then the YAML config part in `async_setup_entry` could be removed at least.